src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2185,6 +2185,15 @@ void os::Linux::print_uptime_info(outputStream* st) {
   }
 }
 
+static void print_container_helper(outputStream* st, jlong j, const char* metrics) {
+  st->print("%s: ", metrics);
+  if (j > 0) {
+    st->print_cr(UINT64_FORMAT " k", uint64_t(j) / 1024);
```

Comment:
Aren't these sizes page-aligned anyway?

Reply:
I don't see why. For example value `OSContainer::memory_limit_in_bytes()` comes from the cgroup interface files. Basically whatever the user specifies via the container engine's `--memory` switch. For example `--memory 123412332` is fine.

```
$ sudo podman run --rm -ti --memory 128974848 --memory-swap 128974848 -v $(pwd)/build/linux-x86_64-server-release/images/jdk:/opt/jdk:z fedora:35 /opt/jdk/bin/java -Xlog:os+container=trace -version 2>&1 | grep -C2 memory.limit_in_bytes
[0.001s][trace][os,container] Path to /memory.limit_in_bytes is /sys/fs/cgroup/memory/memory.limit_in_bytes
[0.001s][trace][os,container] Memory Limit is: 128974848
[0.001s][info ][os,container] Memory Limit is: 128974848
```

Reply:
128974848 is still page aligned :-)

But okay, I think you are right and this is more precise. In Metaspace reports, I usually write "<1KB" for these cases (not 0 but smaller than 1K), but switching to bytes is also fine. User has to read carefully though.

Reply:
:) yes, just noticed. Sorry. Something in the kernel seems to page-align those user values, but I don't know if that's guaranteed. Anyhow `< 1k` is fine with me too.

Reply:
So many comments about formatting - so I dare to add mine as well:

- We have these nice global declarations (K = 1024, M = K*K, ...). Why not use these?
- I like small sizes being displayed as "< 1k"
- How about displaying sizes < 10M as k, < 10G as M, and so forth?
 
These comments are no blockers. It's up to Matthias to honour them.

Reply:
Hello, I adjusted the output of small values following Severins suggestion.
Regarding having k and M ,  this might be a good idea but in the "S Y S T E M" part of hs_err/hs_info we use k/kB and no M and I wanted to be consistent with most other current output.